sql/core/src/test/scala/org/apache/spark/sql/MathFunctionsSuite.scala

```diff
@@ -262,6 +262,17 @@ class MathFunctionsSuite extends QueryTest with SharedSparkSession {
     }
   }
 
+  test("SPARK-44973: conv must allocate enough space for all digits plus negative sign") {
+    withSQLConf(SQLConf.ANSI_ENABLED.key -> false.toString) {
+      val df = Seq(
+        ((BigInt(Long.MaxValue) + 1).toString(16)),
+        (BigInt(Long.MinValue).toString(16))
```

Comment:
May I ask what is the target test coverage for this? ` -8000000000000000` is meaningful?
```
scala> BigInt(Long.MinValue)
res0: scala.math.BigInt = -9223372036854775808

scala> BigInt(Long.MinValue).toString(16)
res1: String = -8000000000000000
```

Reply:
Both `8000000000000000` and `-8000000000000000` trigger the `java.lang.ArrayIndexOutOfBoundsException` before the fix.  There's not really a big change in path coverage with the addition of the second test case, but there's also not really a big change in test runtime.  Since both cases were failing before, it seemed an appropriate tradeoff (to me) to test that the fix addressed both cases.

I suppose I'm more concerned about future regressions due to unknown future refactoring of `NumberConverter` and the benefit of documenting this edge case (the documentation of `conv()` is sparse, in the sense that it has no discussion of edge cases or the effect of the internal `long` representation).

Reply:
+1 for the lack of documentation on the Spark side, I think it is historically borrowed from MySQL via Hive.